src/dune_rules/env_node.ml

```diff
@@ -131,6 +137,16 @@ let make ~dir ~inherit_from ~scope ~config_stanza ~profile ~expander
         { warnings = Option.value config.odoc.warnings ~default:warnings })
   in
   let coq = inherited ~field:coq ~root:config.coq (fun x -> x) in
+  let format_config =
+    Memo.lazy_ (fun () ->
+        match config.format_config with
+        | None -> (
+          match inherit_from with
+          | None ->
+            Code_error.raise "format config not taken from default env" []
+          | Some t -> format_config (Memo.Lazy.force t) )
+        | Some x -> x)
```

Comment:
Would be nice to see this short clause on top.

Reply:
Thanks, done.